### PR TITLE
Actor subscription

### PR DESCRIPTION
**Pull request by lustefaniak**

### CODE REVIEW

File: frontend/src/main/scala/akka/viz/frontend/components/components.scala [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package akka.viz.frontend.components
 
 import akka.viz.frontend.FrontendUtil._
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/components/components.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2002050592/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala [hunk 2]
```diff
@@ -1,6 +1,11 @@
 package akka.viz.frontend
 
-import org.scalajs.dom._
+import org.scalajs.dom.Element
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2002050586/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala [hunk 3]
```diff
@@ -12,8 +17,20 @@ object FrontendUtil {
       l.pathname + path
   }
 
-  def actorName(actorRef: String): String = {
-    actorRef.split("/").drop(3).mkString("/").split("#").head
+  @tailrec
+  def findParentWithAttribute(elem: Element, attributeName: String): js.UndefOr[Element] = {
+    if (elem == null || elem.hasAttribute(attributeName))
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Avoid using null](https://www.codacy.com/app/lustefaniak/akka-viz/file/2002050586/issues/source?bid=3121282&fileBranchId=3122179#l22)

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala [hunk 4]
```diff
@@ -2,8 +2,11 @@ package akka.viz.frontend
 
 import akka.viz.frontend.components.{OnOffPanel, MessagesPanel, MessageFilter, ActorSelector}
 import akka.viz.protocol._
+import org.scalajs.dom.ext.KeyCode
 import org.scalajs.dom.html.Input
-import org.scalajs.dom.{onclick => _, raw => _, _}
+import org.scalajs.dom.raw.{HTMLInputElement, KeyboardEvent, MessageEvent}
+import org.scalajs.dom.document
```
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2002050585/issues/source?bid=3121282&fileBranchId=3122179#l8)

**Comment by pkoryzna** [comment]
:+1:

File: frontend/src/main/scala/akka/viz/frontend/components/components.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/components/components.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004063028/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004063022/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala (hunk 4 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004063021/issues/source?bid=3121282&fileBranchId=3122179#l8)

File: frontend/src/main/scala/akka/viz/frontend/components/components.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/components/components.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004071389/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004071383/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala (hunk 4 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendApp.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004071382/issues/source?bid=3121282&fileBranchId=3122179#l8)

File: backend/src/main/scala/akka/viz/events/LightSnapshot.scala [hunk 5]
```diff
@@ -11,8 +11,9 @@ case class LightSnapshot(
     receivedFrom: Set[(String, String)] = Set()
 ) {
 
-  implicit def ref2String(r: ActorRef): String = r.path.toSerializationFormat
-  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (ref2String(pair._1), ref2String(pair._2))
+  import Helpers.actorRefToString
+
+  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (actorRefToString(pair._1), actorRefToString(pair._2))
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/events/LightSnapshot.scala]: Codacy Issue found: [Found unnamed result or parameter](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004142088/issues/source?bid=3121282&fileBranchId=3122179#l16)

File: backend/src/main/scala/akka/viz/events/LightSnapshot.scala [hunk 6]
```diff
@@ -11,8 +11,9 @@ case class LightSnapshot(
     receivedFrom: Set[(String, String)] = Set()
 ) {
 
-  implicit def ref2String(r: ActorRef): String = r.path.toSerializationFormat
-  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (ref2String(pair._1), ref2String(pair._2))
+  import Helpers.actorRefToString
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/events/LightSnapshot.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004142088/issues/source?bid=3121282&fileBranchId=3122179#l14)

File: frontend/src/main/scala/akka/viz/frontend/components/components.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/components/components.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004142126/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004142120/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: backend/src/main/scala/akka/viz/server/Webservice.scala [hunk 7]
```diff
@@ -27,8 +28,8 @@ object ApiMessages {
 
 }
 
-class Webservice(implicit fm: Materializer, system: ActorSystem) {
-
+class Webservice(implicit fm: Materializer, system: ActorSystem) extends SubscriptionSession {
+  import Helpers.actorRefToString
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/server/Webservice.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004142104/issues/source?bid=3121282&fileBranchId=3122179#l32)

Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/server/Webservice.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004287886/issues/source?bid=3121282&fileBranchId=3122179#l32)

File: backend/src/main/scala/akka/viz/events/LightSnapshot.scala [hunk 8]
```diff
@@ -11,8 +11,9 @@ case class LightSnapshot(
     receivedFrom: Set[(String, String)] = Set()
 ) {
 
-  implicit def ref2String(r: ActorRef): String = r.path.toSerializationFormat
-  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (ref2String(pair._1), ref2String(pair._2))
+  import Helpers._
+
+  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (actorRefToString(pair._1), actorRefToString(pair._2))
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/events/LightSnapshot.scala]: Codacy Issue found: [Found unnamed result or parameter](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004287870/issues/source?bid=3121282&fileBranchId=3122179#l16)

File: backend/src/main/scala/akka/viz/events/LightSnapshot.scala [hunk 9]
```diff
@@ -11,8 +11,9 @@ case class LightSnapshot(
     receivedFrom: Set[(String, String)] = Set()
 ) {
 
-  implicit def ref2String(r: ActorRef): String = r.path.toSerializationFormat
-  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (ref2String(pair._1), ref2String(pair._2))
+  import Helpers._
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/events/LightSnapshot.scala]: Codacy Issue found: [Put imports in the beginning of the package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004287870/issues/source?bid=3121282&fileBranchId=3122179#l14)

File: frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala (hunk 2 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/FrontendUtil.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004287902/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: frontend/src/main/scala/akka/viz/frontend/components/components.scala (hunk 1 above)
Comment (lustefaniak) [inline frontend/src/main/scala/akka/viz/frontend/components/components.scala]: Codacy Issue found: [Use brackets to group imports from the same package](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004287908/issues/source?bid=3121282&fileBranchId=3122179#l3)

File: backend/src/main/scala/akka/viz/events/LightSnapshot.scala [hunk 10]
```diff
@@ -11,8 +13,10 @@ case class LightSnapshot(
     receivedFrom: Set[(String, String)] = Set()
 ) {
 
-  implicit def ref2String(r: ActorRef): String = r.path.toSerializationFormat
-  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = (ref2String(pair._1), ref2String(pair._2))
+  implicit def refPair2StringPair(pair: (ActorRef, ActorRef)): (String, String) = {
+    val (actor1, actor2) = pair
```
Comment (lustefaniak) [inline backend/src/main/scala/akka/viz/events/LightSnapshot.scala]: Codacy Issue found: [Field name does not match the regular expression '^[a-z][A-Za-z0-9]*$'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2004467639/issues/source?bid=3121282&fileBranchId=3122179#l17)

**Pull request merged**